source/features/deprioritize-marketplace-link.tsx

```diff
@@ -1,27 +1,29 @@
 import React from 'dom-chef';
 import select from 'select-dom';
 import onetime from 'onetime';
-import domLoaded from 'dom-loaded';
 import elementReady from 'element-ready';
 import * as pageDetect from 'github-url-detection';
 
 import features from '.';
 
-async function init(): Promise<void> {
-	const marketPlaceLink = (await elementReady('.Header-link[href="/marketplace"]'));
-	if (marketPlaceLink) {
-		// The Marketplace link seems to have an additional wrapper that other links don't have https://i.imgur.com/KV9rtSq.png
-		marketPlaceLink.closest('.border-top, .mr-3')!.remove();
-	}
-
-	await domLoaded;
-
-	select.last('.header-nav-current-user ~ .dropdown-divider')!.before(
+function handleMenuOpening({delegateTarget: dropdown}: delegate.Event): void {
+	dropdown.classList.add('rgh-marketplace-link'); // Mark this as processed
+	select.last('.dropdown-divider', dropdown)!.before(
 		<div className="dropdown-divider"/>,
 		<a className="dropdown-item" href="/marketplace">Marketplace</a>
 	);
 }
 
+async function init(): Promise<void> {
+	(await elementReady('.Header-link[href="/marketplace"]'))
+		// The Marketplace link seems to have an additional wrapper that other links don't have https://i.imgur.com/KV9rtSq.png
+		?.closest('.border-top, .mr-3')!.remove();
+
+	(await elementReady('[aria-label="View profile and more"]'))
+		?.closest('details')!
```

Comment:
Actually, is the `?.` needed in either one? If Enterprise doesn't have the link, it should just be excluded

Reply:
I am not sure.

I think it was added because some pages, like the enter password dont have the links. I dont know if its still relevant

Reply:
> like the enter password dont have the links

Let's find out because I don't care about password pages. They have nothing so, if anything, RGH should not run there at all.

Reply:
51c8d6699afb8540304412df194ec262c503ab23

Reply:
So the code should be updated to use a regular if:

If there’s a marketplace link, move it to the dropdown.

Reply:
With a comment